lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -500,29 +501,51 @@ class MongoClientInterface {
                 },
                 upsert: true,
             },
-        }, {
-            updateOne: {
-                // eslint-disable-next-line
-                filter: {
-                    _id: masterKey,
-                    $or: [{
-                        'value.versionId': {
-                            $exists: false,
+        }];
+        if (objVal.isDeleteMarker && params.vFormat === BUCKET_VERSIONS.v1) {
+            ops.push({
+                deleteOne: {
+                    filter: {
+                        _id: masterKey,
+                        $or: [{
+                            'value.versionId': {
+                                $exists: false,
+                            },
                         },
-                    },
-                    {
-                        'value.versionId': {
-                            $gt: objVal.versionId,
+                        {
+                            'value.versionId': {
+                                $gt: objVal.versionId,
+                            },
                         },
+                        ],
```

Comment:
could this be written the other way round?

```
'value.versionId': {
   $or: [
        { $exists: false },
        { $gt: objVal.versionId }
   ]
}
```

It seems easier to read and less redundant...

Reply:
It is easier to read like that i agree, but unfortunately that is not possible as the `$or` operator doesn't apply to single fields, and so for each item we need to specify the field